jenkins/python-setup.sh

```diff
@@ -17,6 +17,12 @@
 # chosen over any installed ones, and it adds `/usr/sbin` to the end of the
 # path for the `ip` command, used by pbench-register-tool.
 
+# Override the inherited HOME (which is a path generally not mapped inside the
+# container) so that pip and setup.py will install to a directory that exists
+# and doesn't require root access (e.g., to create /home/jenkins/.local/)
+export HOME=/tmp/jenkins
+mkdir -p ${HOME}
+
```

Comment:
To use Peter's phrase, I thinking that adding this here violates the "separation of concerns":  this script is intended to run `python setup` and related stuff, and it could be used outside the Jenkins environment...changing `HOME` inside it limits its applicability (or, is likely to surprise someone who uses it directly).

It is easy enough to set `HOME` on the `podman` command in `jenkins/run`.  Creating the directory is trickier, but we could add it to the container recipe so it's always there.  (I have one other point change that I'd like to add to the recipe, so we can easily do this if you're amenable.)